recipes-devtools/rust/rust.inc

```diff
@@ -192,8 +217,27 @@ def llvm_cpu(d):
     except:
         return trans.get(target, "generic")
 
-TARGET_LLVM_CPU="${@llvm_cpu(d)}"
-TARGET_LLVM_FEATURES = "${@llvm_features_from_tune(d)} ${@llvm_features_from_cc_arch(d)}"
+# generates our target CPU value for cross builds
+def llvm_cross_cpu(d):
+    feat = d.getVar('TUNE_FEATURES', True)
+    if not feat:
+        return ""
+    feat = frozenset(feat.split())
+
+    # The only reliable way of finding the cpu is to look in TUNE_FEATURES
+    # which is derived from DEFAULTTUNE
+    if "mips32r2" in feat:
+        return "mips32r2"
+    elif "mips32" in feat:
+        return "mips32"
+
+    # TODO: Proper x86 targets. "generic" is a valid choice.
+    # TODO: Proper ARM cpu selection, for now "generic" is a valid cpu choice.
+    return "generic"
+
+TARGET_LLVM_CPU="${@llvm_cross_cpu(d)}"
```

Comment:
Looks like this makes all x86 targets "generic" where there were previously some specific options. Is that what we want?